services/app-api/src/postgres/contractAndRates/findRateRelatedContracts.test.tsx

```diff
@@ -0,0 +1,192 @@
+                consolidatedStatus: 'RESUBMITTED',
+            },
+            {
+                id: contractB.id,
+                consolidatedStatus: 'SUBMITTED',
+            },
+        ])
+    )
+
+    await unlockTestContract(
+        cmsServer,
+        submittedContractA.id,
+        'unlocked parent contract'
+    )
+
+    const contractC = await createAndUpdateTestContractWithoutRates(stateServer)
+    // link rate contract C
+    must(
+        await stateServer.executeOperation({
+            query: UpdateDraftContractRatesDocument,
+            variables: {
+                input: {
+                    contractID: contractC.id,
+                    lastSeenUpdatedAt: contractC.draftRevision?.updatedAt,
+                    updatedRates: [
+                        {
+                            type: 'LINK',
+                            rateID: rateAID,
+                        },
+                        {
+                            type: 'CREATE',
+                            formData: testRateFormInputData(),
+                        },
+                    ],
+                },
+            },
+        })
+    )
+
+    await submitTestContract(stateServer, contractC.id)
+
+    const unlockedContractB = await unlockTestContract(
+        cmsServer,
+        contractB.id,
+        'unlock to remove rate A'
+    )
+    const rateBID = unlockedContractB.packageSubmissions[0].rateRevisions.find(
+        (rate) => rate.rateID !== rateAID
+    )?.rateID
+
+    if (!rateBID) {
+        throw new Error('Unexpected error: rateBID should exist, but does not.')
+    }
+
+    // Remove rateA from contractB, but keep it unlocked
+    must(
+        await stateServer.executeOperation({
+            query: UpdateDraftContractRatesDocument,
+            variables: {
+                input: {
+                    contractID: unlockedContractB.id,
+                    lastSeenUpdatedAt:
+                    unlockedContractB.draftRevision?.updatedAt,
+                    updatedRates: [
+                        {
+                            type: 'UPDATE',
+                            formData: testRateFormInputData(),
+                            rateID: rateBID,
+                        },
+                    ],
+                },
+            },
+        })
+    )
+
+    rateARelatedStrippedContracts =  must(await findRateRelatedContracts(client, rateAID))
+
+    // expect all contracts to still be related
+    expect(rateARelatedStrippedContracts).toEqual(
+        expect.arrayContaining([
+            {
+                id: submittedContractA.id,
+                consolidatedStatus: 'UNLOCKED',
+            },
+            {
+                id: contractB.id,
+                consolidatedStatus: 'UNLOCKED',
+            },
+            {
+                id: contractC.id,
+                consolidatedStatus: 'SUBMITTED',
+            },
+        ])
+    )
+
+    // resubmit contract B should remove it from the related contracts
```

Comment:
Nice this feels like useful test, basically making sure we don't persist the relationship changes in related contract until submit